cpp/include/cudf/table/row_operators.cuh

```diff
@@ -373,12 +445,14 @@ class row_lexicographic_comparator {
                                table_device_view lhs,
                                table_device_view rhs,
                                order const* column_order         = nullptr,
-                               null_order const* null_precedence = nullptr)
+                               null_order const* null_precedence = nullptr,
+                               bool const accept_equality        = false)
```

Comment:
What is this?

Reply:
This is to return true if two rows are equal? That's not allowed. This is a strict weak comparator. `<=` and `>=` are not strict.

Reply:
The internals to the row operator get a weak ordering -- essentially running a spaceship comparison inside the row operator that is then interpreted. I can either add a way to interpret the weak ordering equal as true to the existing function or wrap it in a totally parallel operator that has the accept_equality hard coded. This seemed like the better approach.

Reply:
Sorry I don't think I phrased this correctly. The basic search pattern of the `row_lexicographic_comparator` is a spaceship operator.
1. Generates `weak_ordering` comparison values for each column left to right.
2. When a non-`weak_ordering::EQUIVALENT` value is found, convert it into a `bool` comparison value reflecting it's status as `greater` or `less`
3. (Originally) If no non-`weak_ordering::EQUIVALENT` is found, the rows are equivalent and the comparison resolves as false.
3. (Proposed) If non non-`weak_ordering::EQUIVALENT` is found, the rows are equivalent. Resolve the `equal` comparison result as directed by the `accept_equality` configuration.

This is necessary because with `IEEE 754` spec, `NAN` comparison makes `A >= B` not equal to `!(A < B)`.

Reply:
I understand. The correct fix here then is to propagate `weak_ordering` up to the return type of `row_lexicographical_comparator`.

That would obviously be a bigger change that would require updating every use of `row_lexicographical_comparator`, so I suggest factoring out the functionality that depends on this change for a later PR.

Reply:
Or just add new `row_less_comparator` and `row_greater_comparator` as a wrapper around `row_lexicographical_comparator` that returns a `weak_ordering` value.